libs/langchain/langchain/chains/graph_qa/cypher.py

```diff
@@ -176,6 +189,10 @@ def _call(
         # Extract Cypher code if it is wrapped in backticks
         generated_cypher = extract_cypher(generated_cypher)
 
+        # Correct Cypher query if enabled
+        if self.cypher_query_corrector:
+            generated_cypher = self.cypher_query_corrector(generated_cypher)
```

Comment:
Should we raise a warning or error if the generated cypher is not correct? 
Should it actually become null, or rather get a warning attached with the issue pointed out?
Perhaps something for later.

Reply:
at the moment it becomes null, and so the context becomes automatically null.
But we only catch if the relationship direction doesn't fit the schema, we don't catch anything else.

I don't really know if raising an issue is the way to go, we can decide along the way